app/controllers/profile_images_controller.rb

```diff
@@ -2,9 +2,9 @@ class ProfileImagesController < ApplicationController
   before_action :require_logged_in
 
   def destroy
-    image = current_user.profile.images.find_by(id: params[:id])
-    image.purge
+    image = current_user!.profile.images.find_by(id: params[:id])
+    image&.purge
```

Comment:
It might cause an error when the target profile image is already
removed.  This makes the deletion robust by giving idempotency.

I found this bug by Steep's hint. This is the power of types :-)